test/test_binary_ufuncs.py

```diff
@@ -275,19 +88,35 @@ def _helper_reference_numerics(expected, actual, msg, exact_dtype, equal_nan=Tru
             else:
                 self.assertEqualHelper(actual, expected, msg, dtype=dtype, equal_nan=equal_nan, exact_dtype=exact_dtype)
 
-        for l, r in tensor_pairs:
-            if dtype is torch.bfloat16:
-                l_numpy = l.cpu().to(torch.float32).numpy()
-                r_numpy = r.cpu().to(torch.float32).numpy()
-            else:
-                l_numpy = l.cpu().numpy()
-                r_numpy = r.cpu().numpy()
+        for sample in gen:
+            # Each sample input acquired from the generator is just one lhs tensor
+            #   and one rhs tensor
+            l = sample.input
+            r = sample.args[0]
+
+            def to_numpy(x):
```

Comment:
why don't you use SampleInput.numpy()? (SampleInput.numpy doesn't handle bfloat16 though)

Reply:
+1 on using `SampleInput.numpy()` here. It probably also should handle bfloat16 so we can use it as general method for reference tests.

Reply:
Sure; I updated the SampleInput class's `.numpy()` method instead